lib/lotus/helpers/video_helper.rb

```diff
@@ -0,0 +1,86 @@
+module Lotus
+  module Helpers
+    module VideoHelper
+      include Lotus::Helpers::HtmlHelper
+
+      # Generates a video tag for the given arguments.
+      #
+      # Contents are automatically escaped.
+      #
+      # @overload link_to(content, url, options)
+      #   Use string as content
+      #   @param content [String] content used in the a tag
+      #   @param url [String] url used in href attribute
+      #   @param options [Hash] HTML attributes to pass to the a tag
+      #
+      # @overload link_to(url, options, &blk)
+      #   Use block as content
+      #   @param url [String] url used in href attribute
+      #   @param options [Hash] HTML attributes to pass to the a tag
+      #   @param blk [Proc] A block that describes the contents of the a tag
+      #
+      # @return [String] HTML markup for the video
+      #
+      # @raise [ArgumentError] if the signature isn't respected
+      #
+      # @since 0.5.0
+      #
+      # @see Lotus::Helpers::HtmlHelper#html
+      #
+      # @example Basic usage
+      #   <%= video('movie.mp4') %>
+      #     # => <video src="/assets/movie.mp4"></video>
+      #
+      # @example HTML attributes
+      #   <%= video('movie.mp4', autoplay: true, controls: true) %>
+      #     # => <video src="/assets/movie.mp4" autoplay="autoplay" controls="controls"></video>
+      #
+      # @example Fallback Content
+      #   <%=
+      #     video('movie.mp4') do
+      #       "Your browser does not support the video tag"
+      #     end
+      #   %>
+      #     # => <video src="/assets/movie.mp4">\nYour browser does not support the video tag\n</video>
+      #
+      # @example Tracks
+      #   <%=
+      #     video('movie.mp4') do
+      #       track kind: 'captions', src: view.asset_path('movie.en.vtt'), srclang: 'en', label: 'English'
+      #     end
+      #   %>
+      #     # => <video src="/assets/movie.mp4">\n<track kind="captions" src="/assets/movie.en.vtt" srclang="en" label="English">\n</video>
+      #
+      # @example Sources
+      #   <%=
+      #     video do
+      #       text "Your browser does not support the video tag"
+      #       source src: view.asset_path('movie.mp4'), type: 'video/mp4'
+      #       source src: view.asset_path('movie.ogg'), type: 'video/ogg'
+      #     end
+      #   %>
+      #     # => <video>\nYour browser does not support the video tag\n<source src="/assets/movie.mp4" type="video/mp4">\n<source src="/assets/movie.ogg" type="video/ogg">\n</video>
+      #
+      # @example Without any argument
+      #   <%= video %>
+      #     # => ArgumentError
+      #
+      # @example Without src and without block
+      #   <%= video(content: true) %>
+      #     # => ArgumentError
+      def video(src = nil, options = {}, &blk)
+        if src.is_a?(String)
```

Comment:
@leighhalliday For compatibility with other libs it's counterproductive to use `is_a?`. Let's say some other lib has an object called `Options` which can be serialized as a Hash. With this implementation we'll totally ignore that type. This doesn't make our code "Open for extension, Closed for modifications".

As a side note:

```ruby
[1] pry(main)> require 'lotus/utils/hash'
=> true
[2] pry(main)> Lotus::Utils::Hash.new.is_a?(Hash)
=> false
```

This code isn't compatible with Lotus itself :wink: 

What if we do:

```ruby
if src.respond_to?(:to_hash)
  options = src.to_hash
else
  options[:src] = asset_path(src)
end
```

That will make the implementation compatible with `Hash`, `Lotus::Utils::Hash`, any eventual object that can serialize itself into a Hash.

In the `else` condition, we'll leave delegate `asset_path` to coerce into a string.

We don't use `merge` here, which allocates a new Hash that is not really needed. See https://speakerdeck.com/sferik/writing-fast-ruby :wink: